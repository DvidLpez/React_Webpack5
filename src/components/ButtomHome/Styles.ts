import styled from "styled-components";
import { Wrapper } from "../../components/StyledComponents/styledComponents";

export const WrapperLink = styled(Wrapper)`
  padding-top: ${(props) => props.theme.spaces.s20};
  text-align: center;
`;


