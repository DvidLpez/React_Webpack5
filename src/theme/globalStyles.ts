
import { createGlobalStyle } from "styled-components";
import { theme } from "./theme";


export const GlobalStyle = createGlobalStyle`

   html {
      box-sizing: border-box;
   }
   body {
         background-color: ${theme.colors.tertiary};
         font-family: ${theme.fonts.style};
         padding: ${theme.spaces.none};
         margin: ${theme.spaces.none};
   }
   *,
   ::after,
   ::before {
      box-sizing: inherit;
   }

   a{
      text-decoration: none;
      color: ${theme.colors.primary};
   }

   hr{
      border: none;
      border-bottom: solid 1px ${theme.colors.secondary};
   }

   .icon_brand {
      font-size: ${theme.fonts.big};
      margin-right: ${theme.spaces.s15};
      color: ${theme.colors.title};
   }
   .icon_sweat {
      font-size: ${theme.fonts.extra};
   }

   .icon_error {
         font-size: ${theme.fonts.extraBig};
      }

   @media screen and (min-width: 768px) {
   
      .icon_brand {
         font-size: ${theme.fonts.extra};
      }
      
   }
`;
