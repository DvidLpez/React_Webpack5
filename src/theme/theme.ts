import { DefaultTheme } from "styled-components";

export const theme: DefaultTheme = {
  fonts: {
    style: "'Source Sans Pro', sans-serif",
    small: "0.9em",
    default: "1em",
    medium: "1.2em",
    large: "1.6em",
    big: "2em",
    extra: "3em",
    extraBig: "5em",
    bold: "600",
  },
  colors: {
    primary: "#000",
    secondary: "#dbdbdb",
    tertiary: "#fafafa",
    grey: "#414141",
    lightgrey: "#7c7c7c",
    gradient:
      "linear-gradient(to right, #3e9e7e 0%, #2e755d 10%, #205040 20%, #414141 30%)",
    title: "#fff",
    success: "#008000",
    closed: "#c60404",
    danger: "#932323",
    backComment: "#fff9d6",
    shadow: "2px 4px 11px 1px rgba(138, 138, 138, 1)",
  },
  spaces: {
    none: "0",
    center: "center",
    s5: "5px",
    s10: "10px",
    s15: "15px",
    s20: "20px",
    s25: "25px",
    s30: "30px",
    s50: "50px",
    p55: "5px 5px",
    p100: "10px 0",
    p1520: "15px 20px",
    p1010: "10px 10px",
    p1515: "15px 15px",
    p2020: "20px 20px",
    p2525: "25px 25px",
    p3030: "30px 30px",
    button: "10px 40px",
    borderRadius: "5px",
  }
};
