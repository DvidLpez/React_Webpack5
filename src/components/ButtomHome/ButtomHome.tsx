import React, { FC, ReactElement } from "react";
import { LinkHome } from "../StyledComponents/styledComponents";
import { WrapperLink } from "./Styles";


const ButtomHome: FC = (): ReactElement => {

   return (
      <WrapperLink>
         <LinkHome  to="/">
            Go Home
         </LinkHome>
      </WrapperLink> 
   )
}
export default ButtomHome;
