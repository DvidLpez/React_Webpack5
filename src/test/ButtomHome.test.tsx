import React from "react";
import ButtomHome from "../components/ButtomHome/ButtomHome";
import { mount, shallow, render} from 'enzyme';
import { ThemeProvider } from "styled-components";
import { theme } from "../theme/theme";
import { MemoryRouter } from "react-router-dom";


describe('TESTING HOME BUTTOM', () => {

   it('- Check url, value and HTML output', () => {
      const wrapper = render(
         <ThemeProvider theme={theme}>
            <MemoryRouter>
               <ButtomHome />
            </MemoryRouter>
         </ThemeProvider >
      );

      let anchor = wrapper.find('a'),
      url = anchor.attr('href'),
      value = anchor.html();

      expect(url).toEqual('/');   // OK
      expect(value).toMatch('Go Home');  // OK
      expect(wrapper).toMatchSnapshot();  // OK
   });
});
