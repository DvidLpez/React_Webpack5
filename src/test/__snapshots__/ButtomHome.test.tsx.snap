// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTING HOME BUTTOM - Check url, value and HTML output 1`] = `
<div
  class="sc-bdfBwQ sc-fubCfw kUdDos fwLbUW"
>
  <a
    class="sc-gKsewC fMfLPa"
    href="/"
  >
    Go Home
  </a>
</div>
`;
